### PR TITLE
[SPARK-25068][SQL] Add exists function.

### PR DESCRIPTION
**Pull request by ueshin**
## What changes were proposed in this pull request?

This pr adds `exists` function which tests whether a predicate holds for one or more elements in the array.

```sql
> SELECT exists(array(1, 2, 3), x -> x % 2 == 0);
 true
```

## How was this patch tested?

Added tests.

### CODE REVIEW

**Comment by ueshin** [comment]
cc @hvanhovell @gatorsmile

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala [hunk 1]
```diff
@@ -356,6 +356,52 @@ case class ArrayFilter(
...
+  @transient lazy val LambdaFunction(_, Seq(elementVar: NamedLambdaVariable), _) = function
+
+  override def nullSafeEval(inputRow: InternalRow, value: Any): Any = {
+    val arr = value.asInstanceOf[ArrayData]
+    val f = functionForEval
+    var i = 0
+    while (i < arr.numElements) {
+      elementVar.value.set(arr.get(i, elementVar.dataType))
+      if (f.eval(inputRow).asInstanceOf[Boolean]) {
+        return true
```
Comment (cloud-fan) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/higherOrderFunctions.scala]: shall we use a `var exists = false` to keep the result, and stop the loop when result is true `while (i < arr.numElements & !exists)`?

IIUC return in Scala is implemented by throwing an exception, which may have performance issue.

**Comment by cloud-fan** [comment]
LGTM

**Review by hvanhovell — APPROVED** [review]
LGTM

**Review by gatorsmile — COMMENTED** [review]
Thanks! Merged to master.

**Pull request closed without merging**